### PR TITLE
Bump to flink 1.19

### PR DESCRIPTION
**Pull request by henneberger**

### CODE REVIEW

File: sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/DAGPlannerTest/distinctTest.txt [hunk 1]
```diff
@@ -159,7 +159,7 @@ Schema:
 Plan:
 LogicalProject(customerid=[$0], timestamp=[$4], name=[$2])
   LogicalFilter(condition=[=($5, 1)])
-    LogicalProject(customerid=[$0], email=[$1], name=[$2], lastUpdated=[$3], timestamp=[$4], _rownum=[ROW_NUMBER() OVER (PARTITION BY $0 ORDER BY $4 DESC)])
+    LogicalProject(customerid=[$0], email=[$1], name=[$2], lastUpdated=[$3], timestamp=[$4], _rownum=[ROW_NUMBER() OVER (PARTITION BY $0 ORDER BY $4 DESC NULLS FIRST)])
```
Comment (henneberger) [inline sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/DAGPlannerTest/distinctTest.txt:162]: Hm, seems like a new default?

File: sqrl-testing/sqrl-integration-tests/src/test/java/com/datasqrl/DAGPlannerTest.java [hunk 2]
```diff
@@ -21,41 +18,211 @@ protected DAGPlannerTest() {
...
   public Predicate<Path> getDeployDirFilter() {
-    return (p)->false;
+    return (p) -> false;
   }
 
-  static class DagPlannerSQRLFiles extends SqrlScriptArgumentsProvider {
-    public DagPlannerSQRLFiles() {
-      super(SCRIPT_DIR, true);
+  private void testScript(Path script) {
+    System.out.println("PATH: " + script);
```
Comment (henneberger) [inline sqrl-testing/sqrl-integration-tests/src/test/java/com/datasqrl/DAGPlannerTest.java:38]: ill remove

File: sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/DAGPlannerTest/distinctTest.txt (hunk 1 above)
Comment (mbroecheler) [reply sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/DAGPlannerTest/distinctTest.txt:162]: NULLS FIRST is the default, I believe. Are they making it explicit in Calcite now?

File: sqrl-flink-lib/sqrl-vector/src/main/java/com/datasqrl/vector/OnnxEmbed.java [hunk 3]
```diff
@@ -38,6 +48,13 @@ public FlinkVectorType eval(String text, String modelPath) {
     }
   }
 
+  @Override
+  public TypeInference getTypeInference(DataTypeFactory typeFactory) {
```
Comment (henneberger) [inline sqrl-flink-lib/sqrl-vector/src/main/java/com/datasqrl/vector/OnnxEmbed.java:52]: new issue in flink 1.19 w/ udt, need to verify json

File: sqrl-flink-lib/sqrl-vector/src/main/java/com/datasqrl/vector/OnnxEmbed.java [hunk 4]
```diff
@@ -4,18 +4,28 @@
...
 import org.tribuo.interop.onnx.extractors.Tokenizer;
 
 /**
  * Computes a vector embedding for the given string based on the provided ONNX vector embedding model.
  * The embedding model should be stored in a directory that is accessible at runtime.
  */
+@FunctionHint(
+    input = {@DataTypeHint("String"), @DataTypeHint("String")},
+    output = @DataTypeHint("FlinkVectorType")
+)
```
Comment (henneberger) [inline sqrl-flink-lib/sqrl-vector/src/main/java/com/datasqrl/vector/OnnxEmbed.java:28]: will remove

File: sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/DAGWriterJsonTest/clickstream-teaser-clickstream-teaser-package.txt [hunk 5]
```diff
@@ -28,7 +28,7 @@
   "stage" : "flink",
   "inputs" : [ "click_1" ],
   "plan" : "LogicalTableScan(table=[[click_1]], hints=[[[WatermarkHint inheritPath:[] options:[1]]]]) hints[WatermarkHint options:[1]]\n",
-  "sql" : "SELECT *\nFROM `click_1`",
+  "sql" : "SELECT *\nFROM `click_1`\n/*+ `WatermarkHint` */",
```
Comment (mbroecheler) [inline sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/DAGWriterJsonTest/clickstream-teaser-clickstream-teaser-package.txt:31]: I shouldn't be printing the hint here. I wonder if something changed in the SQL printer for Calcite.

File: sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/UseCaseCompileTest/snowflake--package.txt [hunk 6]
```diff
@@ -101,7 +101,7 @@ LogicalTableScan(table=[[myicebergtable_1]])
         },
         {
           "name" : "Query.MyIcebergTable-8",
-          "sql" : "CREATE OR REPLACE VIEW Query.MyIcebergTable-8(id, customer_id, loan_type_id, amount, duration, application_date, updated_at, __pk)\nAS SELECT id, customer_id, loan_type_id, amount, duration, application_date, updated_at, 1 AS __pk\nFROM myicebergtable_1\nWHERE id = ? AND updated_at = ?\nORDER BY updated_at DESC NULLS LAST, 1"
+          "sql" : "CREATE OR REPLACE VIEW Query.MyIcebergTable-8(id, customer_id, loan_type_id, amount, duration, application_date, updated_at, __pk)\nAS SELECT id, customer_id, loan_type_id, amount, duration, application_date, updated_at, 1 AS __pk\nFROM myicebergtable_1\nWHERE id = ? AND updated_at = ?\nORDER BY updated_at DESC NULLS LAST, '1'"
```
Comment (mbroecheler) [inline sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/UseCaseCompileTest/snowflake--package.txt:104]: What's that '1' there at the end. That seems redundant. Not a new issue. Just noticed it.

Comment (henneberger) [reply sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/UseCaseCompileTest/snowflake--package.txt:104]: this looks like an issue, it should ordinal 1 not a char literal

Comment (henneberger) [reply sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/UseCaseCompileTest/snowflake--package.txt:104]: I had to fix it for postgres, ill see if theres a more generic way. I'm not sure why calcite is now doing this. https://github.com/apache/calcite/blob/a6a1e2cef332893fd90286098869c56529e052c3/core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java#L610

**Review by mbroecheler — APPROVED** [review]
2 minor comments. Didn't look too closely at the Calcite stuff. Snapshots looks correct

File: sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/UseCaseCompileTest/snowflake--package.txt (hunk 6 above)
Comment (henneberger) [reply sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/UseCaseCompileTest/snowflake--package.txt:104]: Actually, it should not be an ordinal and we've been introducing an extra incorrect sort (?). In this query we have a '1' as the primary key, but we've been adding it as an ordinal in postgres

Comment (henneberger) [reply sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/UseCaseCompileTest/snowflake--package.txt:104]: This an issue in postgres as it doesn't support sorting by char literals

Comment (mbroecheler) [reply sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/UseCaseCompileTest/snowflake--package.txt:104]: We add a 1 literal as the primary key when a table doesn't have a primary key (e.g. when you aggregate globally without a group by). Not sure why this is happening here. Let me take a look.

File: sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/DAGPlannerTest/streamStateAggregateTest.txt [hunk 7]
```diff
@@ -300,7 +300,7 @@ LogicalTableScan(table=[[product_1]], hints=[[[WatermarkHint inheritPath:[] opti
             "query" : {
               "type" : "PagedJdbcQuery",
               "type" : "PagedJdbcQuery",
-              "sql" : "SELECT \"orders_2\".\"customerid\" AS \"customer\", COUNT(*) AS \"order_count\", MAX(\"orders_2\".\"time\") AS \"__timestamp\"\nFROM \"orders_2\"\n INNER JOIN \"customer_2\" ON \"orders_2\".\"customerid\" = \"customer_2\".\"customerid\"\nGROUP BY \"orders_2\".\"customerid\"\nHAVING \"orders_2\".\"customerid\" = $1",
+              "sql" : "SELECT \"orders_2\".\"customerid\" AS \"customer\", COUNT(*) AS \"order_count\", MAX(\"orders_2\".\"time\") AS \"__timestamp\"\nFROM \"orders_2\"\n INNER JOIN \"customer_2\" ON \"orders_2\".\"customerid\" = \"customer_2\".\"customerid\"\nGROUP BY \"orders_2\".\"customerid\"\nHAVING \"customer\" = $1",
```
Comment (henneberger) [inline sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/DAGPlannerTest/streamStateAggregateTest.txt:303]: postgres doesn't allow having aliases

**Pull request merged**